charts/sqnc-node/templates/tests/checkIfValidator.yaml

```diff
@@ -0,0 +1,104 @@
+          START_BLOCK_LOGS=`curl -sS -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "chain_getHeader", "params": ['$START_HEAD']}' "http://{{ $fullname }}-{{ $i }}:9944" | jq '.result.digest.logs'`
+          START_BLOCK_AUTHOR=$(echo "$START_BLOCK_LOGS" | cut -d 'x' -f 2 | sed -r 's/(.{2})/\\x&/g')
+          if [ "${START_BLOCK_AUTHOR##*"BABE"*}" ]; then
+            echo "caught BABE as author for block $START_HEAD"
+          else
+            echo "failed to match expected block author"
+            exit
+          fi
+
+          # Finalisation check
+          CURRENT_HEAD=$START_HEAD
+
+          echo "block head at start: $START_HEAD"
+
+          count=$((0)); interval=5; timeout=120
+          while [ $CURRENT_HEAD = $START_HEAD ]; do
+            sleep "$interval"s
+
+            CURRENT_HEAD=`curl -sS -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "chain_getFinalizedHead", "params": []}' "http://{{ $fullname }}-{{ $i }}:9944" | jq '.result'`
+
+            echo "block head after ${interval}s: $CURRENT_HEAD"
+
+            count=$((count+interval))
+
+            if [ $count -ge $timeout ]; then
+              echo "block finalisation timeout after $count seconds; no change detected"
+              break
+            else
+              echo "block finalisation witnessed after $count seconds"
+            fi
+          done
+    {{- end }}
+    - name: block-height-check
+      image: {{ $.Values.osShell.image.repository }}:{{ $.Values.osShell.image.tag }}
+      command: [ "/bin/sh" ]
+      args:
+        - -c
+        - |
+          sleep 30
+          SYNC_STATE=`curl -sS -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "system_syncState", "params": []}' "http://{{ $fullname }}-{{ $i }}:9944" | jq '.result'`
+
+          eval "$(echo $SYNC_STATE | jq -r 'to_entries | map("\(.key | ascii_upcase)=\(.value | @sh)") | .[]')"
+
+          BLOCK_DIFF=$(($HIGHESTBLOCK - $STARTINGBLOCK))
+
+          if [ "$STARTINGBLOCK" = "$HIGHESTBLOCK" ]; then
+            echo "block height failed to increase; started at $STARTINGBLOCK and still at $HIGHESTBLOCK"
+            exit
+          else
+            echo "block height is now $HIGHESTBLOCK, having grown by $BLOCK_DIFF"
+          fi
+    - name: node-connection-check
+      image: {{ $.Values.osShell.image.repository }}:{{ $.Values.osShell.image.tag }}
+      command: [ "/bin/sh" ]
+      args:
+        - -c
+        - |
+          SYSTEM_HEALTH=`curl -sS -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "system_health", "params": []}' "http://{{ $fullname }}-{{ $i }}:9944" | jq '.result'`
+
+          eval "$(echo $SYSTEM_HEALTH | jq -r 'to_entries | map("\(.key | ascii_upcase)=\(.value | @sh)") | .[]')"
+
+          if [ "$SHOULDHAVEPEERS" = "true" ]; then
+            echo "expecting that peers connect to the node"
+            if [ "$PEERS" = 0 ]; then
```

Comment:
The value for 0 should be templated in.

We want to set this as `0` when we are doing the helm testing in `ci/ct-values.yaml` and a default of 2 when we are running it elsewhere.